core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -468,8 +466,9 @@ private[spark] class TaskSchedulerImpl(
       resourceProfileIds: Array[Int],
       availableCpus: Array[Int],
       availableResources: Array[Map[String, Buffer[String]]],
-      rpId: Int): Int = {
-    val resourceProfile = sc.resourceProfileManager.resourceProfileFromId(rpId)
+      taskSet: TaskSetManager): Int = {
+    val resourceProfile = sc.resourceProfileManager.resourceProfileFromId(
+      taskSet.taskSet.resourceProfileId)
     val offersForResourceProfile = resourceProfileIds.zipWithIndex.filter { case (id, _) =>
```

Comment:
Any particular reason to change this ?

Reply:
we don't need to pass the raid in anymore since its in the task set which we need now

Reply:
True, but I was trying to make sense of whether it was relevant to the fix or not.
Looks like an unrelated cleanup

Reply:
We need `taskSet: TaskSetManager` now because we'll use it to abort the task set below.

Reply:
Ah ! Yes, I knew I missed something :-)
Thanks